ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -443,11 +446,11 @@ mkApps
      )
   => NodeKernel m remotePeer localPeer blk -- ^ Needed for bracketing only
   -> Tracers m remotePeer blk e
-  -> Codecs blk e m bCS bCS bBF bBF bTX bTX2 bKA
+  -> (NodeToNodeVersion -> Codecs blk e m bCS bCS bBF bBF bTX bTX2 bKA)
   -> m ChainSyncTimeout
   -> Handlers m remotePeer blk
   -> Apps m remotePeer bCS bBF bTX bTX2 bKA ()
-mkApps kernel Tracers {..} Codecs {..} genChainSyncTimeout Handlers {..} =
```

Comment:
Minor stylistic suggestion: what do you think about instead adding a where clause:

```
   Codecs {..} = mkCodecs version
```

EG you wouldn't have to change all the use sites.

Reply:
Ah, nevermind: I was confused about scope.